File: Python/6kyu/Who Likes It/solution.py
def likes(names):
    text = ""
    last_elem = len(names)-1
    
    if not names:
        return "no one likes this"
    
    if len(names) == 1:
        return names[0] + " likes this"
    
    for i, name in enumerate(names):
        if last_elem < 3:
            if i == last_elem:
                text = text + " and " + name + " like this"
            elif i:
                text = text + ", " + name
            else:
                text = name
        else:
            total = last_elem-1
            return names[0] + ", " + names[1] + " and " + str(total) + " others like this"


    return text

File: PHP/6kyu/Who Likes It/solution.php
<?php
function likes( $names ) {
  switch(count($names)) {
    case 0: return "no one likes this"; break;
    case 1: return sprintf("%s likes this", $names[0]); break;
    case 2: return sprintf("%s and %s like this", $names[0], $names[1]); break;
    case 3: return sprintf("%s, %s and %s like this", $names[0], $names[1], $names[2]); break;
    default: return sprintf("%s, %s and %d others like this", $names[0], $names[1], count($names)-2); break;
  }
}
?>